vcd/resource_vcd_vm_sizing_policy.go

```diff
@@ -0,0 +1,604 @@
+							Description:  "Defines the CPU limit in MHz for a VM. If not defined in the VDC compute policy, CPU limit is equal to the vCPU speed multiplied by the number of vCPUs.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+						"shares": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Defines the number of CPU shares for a VM. Shares specify the relative importance of a VM within a virtual data center. If a VM has twice as many shares of CPU as another VM, it is entitled to consume twice as much CPU when these two virtual machines are competing for resources. If not defined in the VDC compute policy, normal shares are applied to the VM.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+					},
+				},
+			},
+			"memory": &schema.Schema{
+				Optional: true,
+				MinItems: 0,
+				MaxItems: 1,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"size_in_mb": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Defines the memory configured for a VM in MB. This is a VM hardware configuration. When a tenant assigns the VM sizing policy to a VM, the VM receives the amount of memory defined by this attribute.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+						"reservation_guarantee": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Defines the reserved amount of memory that is configured for a VM. The value of the attribute ranges between 0 and one. Value of 0 memory reservation guarantee defines no memory reservation. Value of 1 defines 100% of memory reserved.",
+							ValidateFunc: IsFloatAndBetween(0, 1),
+						},
+						"limit_in_mb": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Defines the memory limit in MB for a VM. If not defined in the VM sizing policy, memory limit is equal to the allocated memory for the VM.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+						"shares": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Description:  "Defines the number of memory shares for a VM. Shares specify the relative importance of a VM within a virtual data center. If a VM has twice as many shares of memory as another VM, it is entitled to consume twice as much memory when these two virtual machines are competing for resources. If not defined in the VDC compute policy, normal shares are applied to the VM.",
+							ValidateFunc: IsIntAndAtLeast(0),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVmSizingPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	policyName := d.Get("name").(string)
+	log.Printf("[TRACE] VM sizing policy creation initiated: %s", policyName)
+
+	vcdClient := meta.(*VCDClient)
+
+	if !vcdClient.Client.IsSysAdmin {
+		return fmt.Errorf("functionality requires system administrator privileges")
```

Comment:
Nit, maybe capitalize for easier scanning:
```suggestion
		return fmt.Errorf("functionality requires System Administrator privileges")
```

Reply:
changed to whole project